lib/archethic.ex

```diff
@@ -74,25 +73,50 @@ defmodule Archethic do
     end
   end
 
-  defp forward_transaction(
-         tx,
-         welcome_node_key,
-         nodes \\ P2P.authorized_and_available_nodes()
-         |> Enum.filter(&Node.locally_available?/1)
-         |> P2P.nearest_nodes()
-       )
+  @spec forward_transaction(tx :: Transaction.t(), welcome_node_key :: Crypto.key()) :: :ok
+  defp forward_transaction(tx, welcome_node_key) do
+    %Node{network_patch: welcome_node_patch} = P2P.get_node_info!(welcome_node_key)
+
+    nodes =
+      P2P.authorized_and_available_nodes()
+      |> Enum.reject(&(&1.first_public_key == welcome_node_key))
+      |> Enum.sort_by(& &1.first_public_key)
+      |> P2P.nearest_nodes(welcome_node_patch)
+      |> Enum.filter(&Node.locally_available?/1)
+
+    nodes =
+      if Crypto.first_node_public_key() != welcome_node_key do
+        #  if this node is not the welcome node then select
+        # next node from the this node position in nodes list
+        current_node = Crypto.first_node_public_key()
+
+        index = Enum.find_index(nodes, &(&1.first_public_key == current_node))
```

Comment:
since there is a sort by nearest_nodes, is the order of the nodes determinist?

Reply:
The sort is based on the network patch if the welcome node, not the local node one.
So as the network patch of the welcome is the same across all node, the sort result in the same ordre

Reply:
Yes sure, I didn't see the welcome_node_patch, so definitively as long as the network remains the same across the nodes, it should be deterministic.